### PR TITLE
[POAE7-2810] Migrate non-groupby CiderAggTest to nextgen

### PR DESCRIPTION
**Pull request by imajiayu**
### What changes were proposed in this pull request?
1. Support comparison between `float` and `double` in `CiderArrowChecker`
2. Specially reconstruct non groupby results for PartialAVG
3. migrate&pass all non groupby UTs
4. fix some bug in Aggregate codegen stage

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
CiderAggNextgenTest

### CODE REVIEW

File: cpp/src/cider/exec/nextgen/context/RuntimeContext.cpp [hunk 1]
```diff
@@ -22,6 +22,7 @@
 #include "exec/nextgen/context/RuntimeContext.h"
 #include "exec/module/batch/CiderArrowBufferHolder.h"
 #include "exec/nextgen/operators/extractor/AggExtractorBuilder.h"
+#include "tests/utils/ArrowArrayBuilder.h"
```
Comment (yma11) [inline cpp/src/cider/exec/nextgen/context/RuntimeContext.cpp:25]: Possible to move this `ArrowArrayBuilder.h ` out of  `test_util`? It doesn't make sense for nextgen to depend on test package. I think you can put it under `cider/util`.

File: cpp/src/cider/exec/nextgen/context/RuntimeContext.cpp [hunk 2]
```diff
@@ -119,6 +121,35 @@ Batch* RuntimeContext::getNonGroupByAggOutputBatch() {
         ->extract({buf}, arrow_array->children[i]);
   }
 
+  auto partial_avg_count = std::count_if(
```
Comment (yma11) [inline cpp/src/cider/exec/nextgen/context/RuntimeContext.cpp:124]: add some comments here to tell that this is for partial avg result construction.

File: cpp/src/cider/exec/nextgen/operators/RowToColumnNode.cpp [hunk 3]
```diff
@@ -147,6 +147,7 @@ class ColumnWriter {
           context_.getJITFunction()->createLiteral(JITTypeTag::INT64, 4);
       auto new_offset_buffer = allocateRawDataBuffer(1, offsets_bytes_num)
                                    ->castPointerSubType(JITTypeTag::INT32);
+      // init offset[0] = 0;
```
Comment (yma11) [inline cpp/src/cider/exec/nextgen/operators/RowToColumnNode.cpp:150]: revert this line?

File: cpp/src/cider/tests/nextgen/functional/CiderAggNextgenTest.cpp [hunk 4]
```diff
@@ -0,0 +1,968 @@
...
+
+#include <google/protobuf/util/json_util.h>
+#include <gtest/gtest.h>
+#include "tests/utils/ArrowArrayBuilder.h"
+#include "tests/utils/CiderNextgenTestBase.h"
+#include "tests/utils/QueryArrowDataGenerator.h"
+
+using namespace cider::test::util;
+
+class CiderAggArrowTest : public CiderNextgenTestBase {
```
Comment (yma11) [inline cpp/src/cider/tests/nextgen/functional/CiderAggNextgenTest.cpp:30]: Rename it to `NonGroupbyAggTest`?

**Review by yma11 — APPROVED** [review]

**Review by jikunshang — APPROVED** [review]

**Pull request merged**